c2cgeoportal/views/pdfreport.py

```diff
@@ -0,0 +1,169 @@
+            'version': '1.0.0',
+            'request': 'GetFeature',
+            'typeName': layername,
+            'featureid': featureid
+        }
+        srs = self.config.get('srs_name')
+        if srs is not None:
+            params['srsName'] = srs
+
+        mapserv_url = self.request.route_url('mapserverproxy')
+        vector_request_url = mapserv_url + "?" \
+            + "&".join(["%s=%s" % i for i in params.items()])
+
+        backgroundlayers = self.config.get("layers", {}). \
+            get(layername, {}).get('backgroundlayers')
+        if backgroundlayers is None:
+            backgroundlayers = self.config.get('default_backgroundlayers', '""')
+
+        imageformat = self.config.get("layers", {}). \
+            get(layername, {}).get('imageformat')
+        if imageformat is None:
+            imageformat = self.config.get('default_imageformat', 'image/png')
+
+        spec_template = self.config.get("spec_template")
+        if spec_template is None:
+            spec_template = """
+            {{
+                "layout": "{layername}",
+                "outputFormat": "pdf",
+                "attributes": {{
+                    "paramID": {id},
+                    "map": {{
+                        "projection": "{srs}",
+                        "dpi": 254,
+                        "rotation": 0,
+                        "bbox": [0, 0, 1000000, 1000000],
+                        "zoomToFeatures": {{
+                            "zoomType": "center",
+                            "layer": "vector",
+                            "minScale": 25000
+                        }},
+                        "layers": [{{
+                            "type": "gml",
+                            "name": "vector",
+                            "style": {{
+                                "version": "2",
+                                "[1 > 0]": {{
+                                    "fillColor": "red",
+                                    "fillOpacity": 0.2,
+                                    "symbolizers": [{{
+                                        "strokeColor": "red",
+                                        "strokeWidth": 1,
+                                        "type": "point",
+                                        "pointRadius": 10
+                                    }}]
+                                }}
+                            }},
+                            "opacity": 1,
+                            "url": "{vector_request_url}"
+                        }}, {{
+                            "baseURL": "{mapserv_url}",
+                            "opacity": 1,
+                            "type": "WMS",
+                            "serverType": "mapserver",
+                            "layers": [{backgroundlayers}],
+                            "imageFormat": "{imageformat}"
+                        }}]
+                    }}
+                }}
+            }}
+            """
```

Comment:
:( I relay don't like this template ..., why not using only in the case we specify it in the configuration ...
and with: `loads(dumps(self.config.get("spec")) % (...))` ?

I use the % to avoid issue with `{}`